lesson-4/README.md

```diff
@@ -0,0 +1,46 @@
+# Advanced React - Async
+
+## Goal - understand (unidirectional) data flow
+
+## TODO
+
+* [recap] Component's life cycle hooks recap
+* [recap] State & Props
+* [recap] Children + [key](https://stackoverflow.com/questions/28329382/understanding-unique-keys-for-array-children-in-react-js#28329550) (re-render)
+* [recap] Promises + Async
+* [concept] HOC - Higher order components
+* [concept] FAC - Function as children / [Render callback](http://reactpatterns.com/#render-callback)
+* [new] PropTypes - Runtime TypeSafety
```

Comment:
[minor] I have mentioned PropTypes last week so it is not entirely a new concept. Still a good idea to recap